src/plugins/Miner.cpp

```diff
@@ -138,7 +138,17 @@ void DistrubuteMineralWorker(const sc2::Unit* unit_) {
 
 }  // namespace
 
+Miner::Miner() {
+    m_turns_to_skip = 7;
```

Comment:
Style: Initialization should be done like this:
```
Miner::Miner(): m_turns_to_skip(7)
```

Reply:
Static member variables can't use that type of initialization. But if I have to make a Plugin.cpp, then I can make them non-static and use that style. Just trying to avoid a Plugin.cpp.

Reply:
Please, avoid static variables if possible.

Reply:
You say "Plugin should stay pure virtual" but compiler tells me the Constructor cannot be virtual, and only way to avoid a constructor (non-virtual) is to use static variables in Plugin. Or force all inherited classes to not forget to set these Plugin skip variables in inherited constructors, which I don't quite like, as I think they should be initialized to zero by Plugin, so if inherited classes don't need to use/change them, they don't need to set them. Otherwise, the Plugin variables could be used without being initialized, without any compiler errors and leading to behavior where the inherited Plugin OnStep is never called (large uninitialized value) and new authors may be confused on why.

So, what is solution?
1) Use Plugin constructor (non-virtual)
2) Use Plugin static variables
3) Expect all inherited class constructors to set default values of Plugin member variables, with confusing behavior if they forget.

Reply:
Simply make onStep or Destructor pure virtual.

Reply:
Also, it is not a problem for somebody to remember that base class should be initialised somehow. This is very basics of the language and not related to pure virtual class. This approach is taken to mark this class as interface.